experimental/packages/opentelemetry-sdk-metrics-base/src/Meter.ts

```diff
@@ -130,16 +130,21 @@ export class Meter implements metrics.Meter {
    * @param collectionTime the HrTime at which the collection was initiated.
    * @returns the list of {@link MetricData} collected.
    */
-  async collect(collector: MetricCollectorHandle, collectionTime: HrTime): Promise<MetricData[]> {
-    const result = await Promise.all(Array.from(this._metricStorageRegistry.values()).map(metricStorage => {
+  async collect(collector: MetricCollectorHandle, collectionTime: HrTime): Promise<MetricsData> {
```

Comment:
It seems like `InstrumentationLibraryMetrics` is what a Meter represents regarding the export model. Maybe we can return a `Promise<InstrumentationLibraryMetrics>` here?